imageregistry/v1/types.go

```diff
@@ -81,6 +81,7 @@ type ImageRegistrySpec struct {
 	// replicas determines the number of registry instances to run.
 	Replicas int32 `json:"replicas"`
 	// logging is deprecated, use logLevel instead.
+	// +optional
 	Logging int64 `json:"logging"`
```

Comment:
Do we want to have omitempty there?

Reply:
Zero is unfortunately still a valid value. BTW, now that LogLevel has a default value it screw up the whole logic and I gonna have to rethink the PR on CIRO.

Reply:
defaulting to 2 or whatever corresponds to `Normal` would be an option too.

Reply:
@sttts how defaulting works for upgraded clusters? Will we see the default value after upgrades if the previous version didn't have this field? Or we'll see a zero value?

I'm more curious about `logLevel`, we expected to see an empty string there.

Reply:
We will see the default. It is applied when receiving something from a client **and** when reading from etcd.

Reply:
Rubber duck conversation with audience:

`Logging` and `OperatorSpec.LogLevel` have the same meaning, the initial idea was to use `Logging` only when `OperatorSpec.LogLevel` was not set by the user (empty). As now there is no such thing as  _empty_ `OperatorSpec.LogLevel` we kinda hit a dead end. I believe we have two options:

1. Document that `Logging` was deprecated and users should update `OperatorSpec.LogLevel` if they need more verbosity (I don't think we are able to query how many users use this field, gonna try to check it out)
2. Remove the Default on the `OperatorSpec.LogLevel` (aka postpone the migration problem as we eventually gonna have to get rid of `Logging`).

Reply:
3. take maximum of both

Reply:
I don't think this is not an option. We can't know if `logging` was deliberately set to 0 or it is the default integer value.  If set by the user we should use ["error"](https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/resource/podtemplatespec.go#L25-L35) if not then we should be using `LogLevel` field instead.

Reply:
Do we care about logLevels so much? Would ignore the zero here, deprecate and be done with it.

Reply:
@sttts why do we need this defaulter? I'd rather remove this defaulter than invent some weird aggregator for these fields.

Reply:
We discussed it on today's scrum, we'll use maximum of both

Reply:
> why do we need this defaulter?

Why do you build non-future proof APIs ? You should never depend on unspecified vs. default value of other fields.